app/services/samples/clone_service.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Samples
+  # Service used to clone samples
+  class CloneService < BaseProjectService
+    CloneError = Class.new(StandardError)
+
+    def execute(new_project_id, sample_ids)
+      validate(new_project_id, sample_ids)
+
+      authorize! @project, to: :clone_sample?
+
+      @new_project = Project.find_by(id: new_project_id)
+      authorize! @new_project, to: :clone_sample_into_project?
+
+      clone_samples(@new_project, sample_ids)
+    rescue Samples::CloneService::CloneError => e
+      @project.errors.add(:base, e.message)
+      {}
+    end
+
+    private
+
+    def validate(new_project_id, sample_ids)
+      raise CloneError, I18n.t('services.samples.clone.empty_new_project_id') if new_project_id.blank?
+
+      raise CloneError, I18n.t('services.samples.clone.empty_sample_ids') if sample_ids.blank?
+
+      return unless @project.id == new_project_id.to_i
+
+      raise CloneError, I18n.t('services.samples.clone.same_project')
+    end
+
+    def clone_samples(new_project, sample_ids) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+      cloned_sample_ids = {}
+      sample_ids.each do |sample_id|
+        sample = Sample.find_by(id: sample_id, project_id: @project.id)
+        clone = Sample.new(name: sample.name, description: sample.description, project_id: new_project.id)
+        metadata_changes = Samples::Metadata::UpdateService.new(new_project, clone, @current_user,
```

Comment:
Can you refactor line 39 through 47 into a new method `update_metadata` which accepts the clone and new_project as params. This should help get rid of the rubocop disables on this method

Reply:
Yup! 5d3d8a64bbb3582a5ef94f9815846603a8994f54 and c5afb29f1d3d500b0a04c57f8663e46717ae8e1e.Yup! 5d3d8a64bbb3582a5ef94f9815846603a8994f54.